src/HDF5.jl

```diff
@@ -2047,7 +2068,12 @@ for (jlname, h5name, outtype, argtypes, argsyms, msg) in
     )
 
     ex_dec = funcdecexpr(jlname, length(argtypes), argsyms)
-    ex_ccall = ccallexpr(libhdf5, h5name, outtype, argtypes, argsyms)
+    library = startswith(string(h5name), "H5DO") ? libhdf5_hl : libhdf5
+    if library == ""
```

Comment:
with the above change this would then be `library === nothing`

Reply:
Maybe it doesn't matter, but I think I wanted the compiler to know that
```
library = startswith(string(h5name), "H5DO") ? libhdf5_hl : libhdf5
```
to always be a string.